Backend/Backend/Services/WalletServices.cs

```diff
@@ -0,0 +1,101 @@
+﻿using Backend.Infrastructure;
+using Backend.Models;
+using Backend.Services.Interfaces;
+using Microsoft.EntityFrameworkCore;
+using Backend.Controllers.Responses;
+using System.Collections;
+
+namespace Backend.Services
+{
+    public class WalletServices : IWalletServices
+    {
+        private readonly AppDbContext _db;
+        private readonly ILogger _logger;
+        public WalletServices(AppDbContext db, ILogger logger)
+        {
+            _db = db;
+            _logger = logger;
+
+        }
+        /// <summary>
+        /// Withdraws Cash From users Wallet
+        /// </summary>
+        /// <param name="userId"></param>
+        /// <param name="amount"></param>
+        /// <returns></returns>
+        /// <exception cref="Exception"></exception>
+        public async Task<bool> Withdraw(Guid userId, double amount)
+        {
+            // Check Amount value is valid or not
+            if (amount <= 0 || amount > 200)
+            {
+                throw new Exception("Withdraw amount should be greater then zero and less then 200");
+
+            }
+            // Check if the user have that much to withdraw
+            double balance = await GetUserBalance(userId);
+
+            // If amount inputted is greater then the balance throw an error
+            if (amount > balance)
+            {
+                throw new Exception("You can't withdraw more then your balance");
+
+            }
+
+            // If all the checks have passed then Proceed with withdrawal
+            await _db.AddAsync(new Transactions() { Amount = -amount,TransactionType = Transactions.Type.Withdraw,UserId = userId, CreatedAt = DateTime.UtcNow });
+
+            // Save the changes to the database
+            return await _db.SaveChangesAsync() > 0;
+
+        }
+        /// <summary>
+        /// Gets the current Balance of the User
+        /// </summary>
+        /// <param name="userId"></param>
+        /// <returns></returns>
+        public async Task<double> GetUserBalance(Guid userId)
+        {
+            return await _db.Transactions.Where(u => u.UserId == userId).SumAsync(u => u.Amount);
```

Comment:
wait, why not save the current balance of the user on the user itself? this feels REALLY convoluted